### PR TITLE
OpenMPTarget: Fix bug in  for the case of a reducer.

### PR DESCRIPTION
**Pull request by rgayatri23**
The PR replace `+` with `join` in case when nested reduction is performed on a reducer.

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp [hunk 1]
```diff
@@ -1641,7 +1641,8 @@ KOKKOS_INLINE_FUNCTION
   for (iType i = loop_boundaries.start; i < loop_boundaries.end; i++) {
     ValueType tmp = ValueType();
```
Comment (masterleinad) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:1642]: This should likely also be replaced by `reducer.init(...)`.

Comment (rgayatri23) [reply core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:1642]: Yes. Its a WIP. Sorry forgot to mark that label. Currently that loop is not encountered though.

**Comment by crtrott** [comment]
What is the status?

**Comment by rgayatri23** [comment]
> What is the status?

This is ready for review.

**Review by crtrott — APPROVED** [review]

**Pull request merged**